Data-Structure-Arrays.md

```diff
@@ -52,9 +52,55 @@ intarray.count(1) # Returns the number of occurrences of 1 in the array i.e. 2
 intarray.insert(1, 3) # Insert a new item with value x in the array before position i
 intarray.remove(1) # Remove the first occurrence of 1 from the array
 intarray.reverse() # Reverse the order of the items in the array
-intarray.pop(1) # Removes the item with the index 1 from the array and returns it 
+intarray.pop(1) # Removes the item with the index 1 from the array and returns it
 ```
 
 :rocket: [Run Code](https://repl.it/CWJB)
 
 [Official Docs](https://docs.python.org/3.5/library/array.html)
+
+## Arrays in C++
+
+C++ provides a data structure, `array`, which stores a fixed-size sequential collection of elements of the same data-type. An `array` is used to store a collection of data, but it is better to think of an `array` as a collection of variables of the same type.
+
+#### Declaration of `array`
+
+```cpp
+int intarray[10]; // Declares an array of integer type of size 10 with elements having random values. Index ranges from 0 to 9(i.e. size-1).
+int intarray[10] = { 0 }; // Declares an array of integer of size 10 with all elements having value 0
+
+// Choose one the two declarations and then move ahead.
+```
+
+#### Adding elements to `array`:
+
+```cpp
+intarray[0] = 1; // Adds an integer value of 1 at index 0
+intarray[1] = 0; // Adds an integer value of 0 at index 1
+intarray[2] = -1; // Adds an integer value of -1 at index 2
+intarray[3] = 1; // Again adds an integer value of 1 at index 3
+
+intarray[4] = "dd"; // Would give a TypeError as the array is of integer type.
+
+// Resolve the above error and then move ahead.
+```
+
+#### Printing an `array`:
+
+```cpp
+std::cout << intarray[0] << '\n'; // Returns 1 which is element at index of the array
+std::cout << intarray[11] << '\n'; // Would give a random value as there is no element at index 11 of array.
```

Comment:
` // Would give a a "Garbage" value as there is no element at index 11 of array, because that memory location is beyond the range of the array.`

Else it sounds like a random no is expected, while in reality this is dangerous, and is primary cause of crashes.
Its accessing a memory location which does not exist.